src/lib/updating-element.ts

```diff
@@ -215,6 +217,33 @@ export abstract class UpdatingElement extends HTMLElement {
 
   static properties: PropertyDeclarations = {};
 
+  /**
+   * Array of styles to apply to the element. The styles should be defined
+   * using the `css` tag function.
+   */
+  static get styles(): CSSResult[] {
+    return [];
+  }
+
+  private static _styles: CSSResult[]|undefined;
+
+  private static get _uniqueStyles(): CSSResult[] {
+    if (this._styles === undefined) {
+      const styles = this.styles;
+      // As a performance optimization to avoid duplicated styling that can
+      // occur especially when composing via subclassing, de-duplicate styles
+      // preserving the last item in the list. The last item is kept to
+      // try to preserve cascade order with the assumption that it's most
+      // important that last added styles override previous styles.
+      const styleSet = styles.reduceRight((set, s) => {
```

Comment:
can just be `styles.reduceRight((set, s) => set.add(s), new Set())`

(set.add returns the set)

Reply:
Done.